### PR TITLE
Add support for protocol buffers (nanopb)

### PR DESCRIPTION
**Pull request by fdcavalcanti**
Hi everyone, please let me know if there's anything I can do to improve this PR.
I'm not sure how this would go regarding the code license.
Thanks!

## Summary
This PR adds support for protocol buffers using [nanopb](https://github.com/nanopb/nanopb).
The Kconfig has an option to enable nanopb and another to compile the example, which is the "simple" example inside the download nanopb source.

## Impact
Adds support for protocol buffers.

## Testing
Tested using the "simple" example in the nanopb source code:
```
nsh> nanopb_example
Your lucky number was 13!
```

### CODE REVIEW

File: netutils/nanopb/Makefile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+CSRCS += $(NANOPB_SRCDIR)$(DELIM)pb_encode.c
+
+ifneq ($(CONFIG_NETUTILS_NANOPB_EXAMPLE),)
+PROGNAME  = nanopb_example
+PRIORITY  = $(CONFIG_NETUTILS_NANOPB_EXAMPLE_PRIORITY)
+STACKSIZE = $(CONFIG_NETUTILS_NANOPB_EXAMPLE_STACKSIZE)
+MODULE    = $(CONFIG_NETUTILS_NANOPB_EXAMPLE)
+
+EXAMPLE_DIR = $(NANOPB_UNPACK)$(DELIM)examples$(DELIM)simple
+MAINSRC   = $(EXAMPLE_DIR)$(DELIM)simple.c
```
Comment (xiaoxiang781216) [inline netutils/nanopb/Makefile:60]:
```suggestion
MAINSRC = $(EXAMPLE_DIR)$(DELIM)simple.c
```

File: netutils/nanopb/Makefile [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	NANOPB_NAME = nanopb-$(NANOPB_VERSION)-windows-x86
+else ifeq ($(CONFIG_HOST_LINUX),y)
+	NANOPB_NAME = nanopb-$(NANOPB_VERSION)-linux-x86
+else
+	NANOPB_NAME = nanopb-$(NANOPB_VERSION)-macosx-x86
+endif
+
+NANOPB_TARBALL  = $(NANOPB_NAME).tar.gz
+NANOPB_UNPACK   = nanopb
+NANOPB_SRCDIR   = $(NANOPB_UNPACK)
```
Comment (xiaoxiang781216) [inline netutils/nanopb/Makefile:34]:
```suggestion
NANOPB_TARBALL = $(NANOPB_NAME).tar.gz
NANOPB_UNPACK  = nanopb
NANOPB_SRCDIR  = $(NANOPB_UNPACK)
```

File: netutils/nanopb/Kconfig [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+if NETUTILS_NANOPB_EXAMPLE
+
+config NETUTILS_NANOPB_EXAMPLE_PRIORITY
+	int "Task Priority"
+	default 100
+
+config NETUTILS_NANOPB_EXAMPLE_STACKSIZE
+	int "Task stack size"
+	default 8192
```
Comment (xiaoxiang781216) [inline netutils/nanopb/Kconfig:37]: can we use the default task stack size?
```suggestion
	default DEFAULT_TASK_STACKSIZE
```

File: netutils/nanopb/Makefile (hunk 2 above)
Comment (fdcavalcanti) [reply netutils/nanopb/Makefile:34]: I don't get it

**Comment by fdcavalcanti** [comment]
I just realized that a bigger issue here is the location. I put this under 'netutils' because I had in mind using it in a socket example. But that's not really the optimal place, is it?

File: netutils/nanopb/Kconfig [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+config NETUTILS_NANOPB
+	tristate "Nanopb Protocol Buffers"
+	default n
+	---help---
+		Nanopb is a small code-size Protocol Buffers implementation in ansi C.
+		It is especially suitable for use in microcontrollers, but fits any
+		memory restricted system.
+		Important: the example option requires Python to be installed
+		and also the following packages:
```
Comment (acassis) [inline netutils/nanopb/Kconfig:14]: "and also the following packages:" I think the packages name are missing

Comment (fdcavalcanti) [reply netutils/nanopb/Kconfig]: Thanks! This line is not needed anymore since the download tarball contains the executables, no Python is necessary.

**Comment by xiaoxiang781216** [comment]
> I just realized that a bigger issue here is the location. I put this under 'netutils' because I had in mind using it in a socket example. But that's not really the optimal place, is it?

how about system folder?

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**